test/optimizer/rule_test.cpp

```diff
@@ -54,5 +69,176 @@ TEST_F(RuleTests, SimpleRuleApplyTest) {
+      "(1,52,'hello_1');");
+
+  statement->SetPlanTree(
+      optimizer::SimpleOptimizer::BuildPelotonPlanTree(insert_stmt));
+
+  result_format =
+      std::move(std::vector<int>(statement->GetTupleDescriptor().size(), 0));
+  status = bridge::PlanExecutor::ExecutePlan(statement->GetPlanTree().get(),
+                                             params, result, result_format);
+  LOG_INFO("Statement executed. Result: %d", status.m_result);
+  LOG_INFO("Tuple inserted!");
+  txn_manager.CommitTransaction(txn);
+
+  // Now Create index
+  txn = txn_manager.BeginTransaction();
+  LOG_INFO("Creating and Index");
+  LOG_INFO("Query: CREATE INDEX saif ON department_table (student_id);");
+  statement.reset(new Statement(
+      "CREATE", "CREATE INDEX saif ON department_table (student_id);"));
+
+  auto update_stmt = peloton_parser.BuildParseTree(
+      "CREATE INDEX saif ON department_table (student_id);");
+
+  statement->SetPlanTree(
+      optimizer::SimpleOptimizer::BuildPelotonPlanTree(update_stmt));
+
+  result_format =
+      std::move(std::vector<int>(statement->GetTupleDescriptor().size(), 0));
+  status = bridge::PlanExecutor::ExecutePlan(statement->GetPlanTree().get(),
+                                             params, result, result_format);
+  LOG_INFO("Statement executed. Result: %d", status.m_result);
+  LOG_INFO("INDEX CREATED!");
+  txn_manager.CommitTransaction(txn);
+
+  auto target_table_ = catalog::Catalog::GetInstance()->GetTableWithName(
+      DEFAULT_DB_NAME, "department_table");
+  // Expected 1 , Primary key index + created index
+  EXPECT_EQ(target_table_->GetIndexCount(), 2);
+
+  // Test update tuple with index scan
+  txn = txn_manager.BeginTransaction();
+  LOG_INFO("Updating a tuple...");
+  LOG_INFO(
+      "Query: UPDATE department_table SET dept_name = 'CS' WHERE student_id = 52");
+  statement.reset(new Statement(
+      "UPDATE",
+      "UPDATE department_table SET dept_name = 'CS' WHERE student_id = 52"));
+  LOG_INFO("Building parse tree...");
+  update_stmt = peloton_parser.BuildParseTree(
+      "UPDATE department_table SET dept_name = 'CS' WHERE student_id = 52");
+  LOG_INFO("Building parse tree completed!");
+  LOG_INFO("Building plan tree...");
+  statement->SetPlanTree(
+      optimizer::SimpleOptimizer::BuildPelotonPlanTree(update_stmt));
+  LOG_INFO("Building plan tree completed!");
+  planner::AbstractPlan *plan = statement->GetPlanTree().get();
+  bridge::PlanExecutor::PrintPlan(statement->GetPlanTree().get(), "Plan");
+
+  // Check scan plan
+  ASSERT_FALSE(plan == nullptr);
+  if (plan->GetPlanNodeType() == PLAN_NODE_TYPE_UPDATE) {
+    auto& scanPlan = plan->GetChildren().front();
+    EXPECT_EQ(scanPlan->GetPlanNodeType(), PLAN_NODE_TYPE_INDEXSCAN);
+  }
+  LOG_INFO("Executing plan...");
+  result_format =
+      std::move(std::vector<int>(statement->GetTupleDescriptor().size(), 0));
+  status = bridge::PlanExecutor::ExecutePlan(statement->GetPlanTree().get(),
```

Comment:
Do not need to execute the plan if we are only testing the optimizer. Limit the scope of the test especially when it is unit test.